bandit/plugins/pytorch_load_save.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024 Stacklok, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+r"""
+=========================================
+B704: Test for unsafe PyTorch load or save
+=========================================
+
+This plugin checks for the use of `torch.load` and `torch.save`. Using `torch.load`
+with untrusted data can lead to arbitrary code execution, and improper use of
+`torch.save` might expose sensitive data or lead to data corruption.
+
+:Example:
+
+.. code-block:: none
+
+        >> Issue: Use of unsafe PyTorch load or save
+        Severity: Medium   Confidence: High
+        CWE: CWE-94 (https://cwe.mitre.org/data/definitions/94.html)
+        Location: examples/pytorch_load_save.py:8
+        7    loaded_model.load_state_dict(torch.load('model_weights.pth'))
+        8    another_model.load_state_dict(torch.load('model_weights.pth', map_location='cpu'))
+        9
+        10   print("Model loaded successfully!")
+
+.. seealso::
+
+     - https://cwe.mitre.org/data/definitions/94.html
+
```

Comment:
Any other articles to reference here? Might be useful to give more context.

Reply:
https://pytorch.org/docs/stable/generated/torch.load.html#torch.load

Reply:
In the case of Hugging Face, they recommend using the alternative of safetensors to torch.load/pickle.

https://github.com/huggingface/safetensors